lib/manageiq/integration/foreman_manager.rb

```diff
@@ -0,0 +1,106 @@
+require "fileutils"
+
+# A wrapper around the foreman utility for running ManageIQ in dev mode for
+module ManageIQ
+  module Integration
+    class ForemanManager
+      def self.pid_file
+        @pid_file ||= Environment.tmp_dir.join('foreman.pid')
+      end
+
+      def self.log_file
+        Rails.root.join("log", "integration.foreman.log")
```

Comment:
by prefix, I mean the `File.basename("..., ".*")` or the basename minus the extension.

Reply:
I avoided a constant for the same reason as this concern here:

https://github.com/ManageIQ/manageiq/pull/20350#discussion_r453811294

So this is my attempt to "lazy load" that call.

Reply:
That said, I don't mind truing this up a bit and having a layer of abstraction in `Environment` as well.

Reply:
I am going to have to sync up on this, because currently I have the prefix for `pid_file` and `log_file` in two different locations.  Using `Rails.root` as a common denominator:

```ruby
pid_file_prefix = Rails.root.join("tmp", "integration")
log_file_prefix = Rails.root.join("log")
```

I think it makes sense to keep the log in the log dir, but maybe you have different thoughts about that.

Reply:
Ok, as long as we can reuse locations and these locations be consistent. It should be easy to track down the pid file and log if we need to.